src/adapter/src/config/params.rs

```diff
@@ -0,0 +1,149 @@
+            .filter(move |var| modified.contains(var.name()))
+            .map(|var| {
+                let name = var.name().to_string();
+                let value = var.value();
+                // This will never panic because both the name and the value come from a `Var` instance.
+                let is_default = self.system_vars.is_default(&name, &value).unwrap();
+                ModifiedParameter {
+                    name,
+                    value,
+                    is_default,
+                }
+            })
+            .collect()
+    }
+
+    /// Get the current in-memory value for the parameter identified by the
+    /// given `name`.
+    ///
+    /// # Panics
+    ///
+    /// The method will panic if the name does not refer to a valid parameter.
+    pub fn get(&self, name: &str) -> String {
+        self.system_vars
+            .get(name)
+            .expect("valid system parameter name")
+            .value()
+    }
+
+    /// Try to modify the in-memory `value` for `name` backing this
+    /// [SynchronizedParameters] instace, calling `SystemVars::reset` iff
+    /// `value` is the default for this `name` and `SystemVars::set` otherwise.
+    ///
+    /// Return `true` iff the backing in-memory value for this `name` has
+    /// changed.
+    pub fn modify(&mut self, name: &str, value: &str) -> bool {
+        // Resolve name to &'static str and assert that the system parameter can
+        // be indeed modified.
+        if let Some(name) = self.synchronized.get(name) {
+            // It's OK to call `unwrap_or(false)` here because for fixed `name`
+            // and `value` an error in `self.is_default(name, value)` implies
+            // the same error in `self.system_vars.set(name, value)`.
+            let modified = if self.system_vars.is_default(name, value).unwrap_or(false) {
+                self.system_vars.reset(name)
+            } else {
+                self.system_vars.set(name, value)
+            };
+            match modified {
+                Ok(true) => {
+                    self.modified.insert(name);
+                    true
+                }
+                Ok(false) => {
+                    // The value was the same as the current one.
+                    false
+                }
+                Err(e) => {
+                    tracing::error!("cannot modify system parameter {}: {}", name, e);
+                    false
+                }
+            }
+        } else {
+            tracing::error!("cannot modify unsynchronized system parameter {}", name);
+            false
+        }
+    }
+}
+
+pub struct ModifiedParameter {
+    pub name: String,
+    pub value: String,
+    pub is_default: bool,
+}
+
+impl ModifiedParameter {
+    pub fn as_alter_system(&self) -> String {
+        match self.is_default {
+            true => format!("ALTER SYSTEM RESET {}", self.name),
+            false => format!("ALTER SYSTEM SET {} = {}", self.name, self.value),
+        }
+    }
```

Comment:
> This should return a statement directly.

I'm not sure things will work out fine with the proposed `as_stmt` for the following query:

```sql
ALTER SYSTEM SET allowed_cluster_replica_sizes = 'xlarge', '2xlarge';
```

The problem is that the `self.value: String` in this case will be `'xlarge', '2xlarge'` and if we pack it a string literal the resulting `sequence_alter_system_set` call will be confused. 

The general problem maybe is that `impl Var for T` provides primarily `T ⇔ String` conversions, whereas it might be better to focus on `T ⇔ Statement<Raw>` conversions and have default implementations `T ⇔ String` that go through `parse()` and `to_string()` of the resulting `Statement`.

I'll try to fix this in #16513 next. Once we should be able to use the extended `Var` API to directly construct the statement as you propose.

Reply:
This actually works (I think), so I accepted the suggestion. I will revert if I encounter errors when adding the CI tests.